src/clan.cpp

```diff
@@ -81,18 +84,18 @@ void do_clantalk(CHAR_DATA *ch, const char *argument) {
 
     /* Next check to see if a CLAN_HERO or CLAN_LEADER is on first */
 
-    for (d = descriptor_list; (d && !candoit); d = d->next) {
-        CHAR_DATA *vix = d->person();
-
-        if (vix && d->character() && vix->pcdata->pcclan
-            && (vix->pcdata->pcclan->clan->clanchar == orig_clan->clan->clanchar)
-            && (vix->pcdata->pcclan->clanlevel >= CLAN_HERO) && !IS_SET(vix->comm, COMM_QUIET))
-            candoit = 1; /* Yeah we can do it! */
-    } /* for all descriptors */
-
-    if (!candoit) {
-        send_to_char("Your clan lacks the necessary broadcast nexus, causing your vain telepathy to\n\rbe lost upon "
-                     "the winds.\n\r",
+    auto playing = descriptors().playing();
+    if (ranges::find_if(playing,
+                        [&](const Descriptor &d) {
+                            const auto *vix = d.person();
```

Comment:
The 'vix' nomenclature seems particular to clan.cpp.

Reply:
side note: I don't know why we require a clan hero/leader to be online for the clan chat to work. Seems novel but could be quite irritating (not that it matters right now!)

Reply:
re: hero online: Yeah it annoyed me during testing. I didn't change existing nomenclature, but will do.